internal/controller/controllers/agent_controller.go

```diff
@@ -77,6 +77,37 @@ type AgentReconciler struct {
 	ApproveCsrsRequeueDuration time.Duration
 }
 
+func (r *AgentReconciler) getClusterDeployment(agent *aiv1beta1.Agent, context *context.Context) (*hivev1.ClusterDeployment, error) {
+	kubeKey := types.NamespacedName{
+		Namespace: agent.Spec.ClusterDeploymentName.Namespace,
```

Comment:
You shouldn't access `agent.Spec.ClusterDeploymentName` before you know it's not `nil`